packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkConfigurationExt.kt

```diff
@@ -75,31 +76,108 @@ internal fun ReadableMap.asProxyConfig(): Pair<Proxy, ProxyAuthenticator?>? {
         port = getInt("port")
     }
 
-    val type =
-        getString("type")?.let {
-            when (it.lowercase(Locale.US)) {
+    val type = getString("type")
+
+    if (address != null && port != null && type != null) {
+        return JSONProxyConfiguration(
+            type,
+            address,
+            port,
+            getString("username"),
+            getString("password")
+        ).asProxyConfig()
+    } else {
+        return null
+    }
```

Comment:
minor suggestion to be more Kotlin-idiomatic

```suggestion
    return if (address != null && port != null && type != null) {
       JSONProxyConfiguration(
            type,
            address,
            port,
            getString("username"),
            getString("password")
        ).asProxyConfig()
    } else {
       null
    }
```